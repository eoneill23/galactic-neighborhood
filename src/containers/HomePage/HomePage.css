.home-container {
  background-color: black;
  display: grid;
  grid-column-gap: 20px;
  grid-template-areas:
    'img' 'asteroids' 'iss'
    'img' 'asteroids' 'iss';
  grid-template-columns: 58% 18% 18%;
  height: 520px;
  padding: 30px;
}

.img-header {
  margin-bottom: 10px;
}

.img-container {
  grid-area: 'img'
}

h2 {
  color: white;
  margin: 0;
}

.asteroids-list {
  color: white;
  grid-area: 'asteroids';
  height: 430px;
  overflow: scroll;
  padding: 0;
}

.iss-container {
  grid-area: 'iss';
}

.apod {
  height: 400px;
  width: 100%;
}

.img-caption {
  text-align: center;
}

.img-copyright {
  font-size: .6em;
  text-align: center;
}

.map-container {
  height: 70px;
  width: 70px;  
}

.map-label {
  color: white;
  font-weight: bold;
  text-decoration: underline;
}