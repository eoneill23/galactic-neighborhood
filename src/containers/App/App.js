import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';
import './App.css';
import Header from '../Header/Header';
import HomePage from '../HomePage/HomePage';
import PageContainer from '../PageContainer/PageContainer';

class App extends Component {
  render() {
    return (
      <>
      <Header />
      <Route exact path="/" render={ () => <HomePage /> } />
      <Route exact path="/mercury" render={ () => <PageContainer 
        fact={this.props.planetFacts['Mercury'].fact}
        imgUrl={this.props.planetFacts['Mercury'].imgUrl}
        factUrl={this.props.planetFacts['Mercury'].factUrl}
        factSite={'Universe Today'}
      /> } />
      <Route exact path="/venus" render={ () => <PageContainer 
        fact={this.props.planetFacts['Venus'].fact}
        imgUrl={this.props.planetFacts['Venus'].imgUrl}
        factUrl={this.props.planetFacts['Venus'].factUrl}
        factSite={'Universe Today'}
      /> } />
      <Route exact path="/earth" render={ () => <PageContainer 
        fact={this.props.planetFacts['Earth'].fact}
        imgUrl={this.props.planetFacts['Earth'].imgUrl}
        factUrl={this.props.planetFacts['Earth'].factUrl}
        factSite={'Universe Today'}
      /> } />
      <Route exact path="/mars" render={ () => <PageContainer 
        fact={this.props.planetFacts['Mars'].fact}
        imgUrl={this.props.planetFacts['Mars'].imgUrl}
        factUrl={this.props.planetFacts['Mars'].factUrl}
        factSite={'Universe Today'}
      /> } />
      <Route exact path="/jupiter" render={ () => <PageContainer 
        fact={this.props.planetFacts['Jupiter'].fact}
        imgUrl={this.props.planetFacts['Jupiter'].imgUrl}
        factUrl={this.props.planetFacts['Jupiter'].factUrl}
        factSite={'Universe Today'}
      /> } />
      <Route exact path="/saturn" render={ () => <PageContainer 
        fact={this.props.planetFacts['Saturn'].fact}
        imgUrl={this.props.planetFacts['Saturn'].imgUrl}
        factUrl={this.props.planetFacts['Saturn'].factUrl}
        factSite={'Universe Today'}
      /> } />
      <Route exact path="/uranus" render={ () => <PageContainer 
        fact={this.props.planetFacts['Uranus'].fact}
        imgUrl={this.props.planetFacts['Uranus'].imgUrl}
        factUrl={this.props.planetFacts['Uranus'].factUrl}
        factSite={'Universe Today'}
      /> } />
      <Route exact path="/neptune" render={ () => <PageContainer 
        fact={this.props.planetFacts['Neptune'].fact}
        imgUrl={this.props.planetFacts['Neptune'].imgUrl}
        factUrl={this.props.planetFacts['Neptune'].factUrl}
        factSite={'Universe Today'}
      /> } />
      <Route exact path="/pluto" render={ () => <PageContainer 
        fact={this.props.planetFacts['Pluto'].fact}
        imgUrl={this.props.planetFacts['Pluto'].imgUrl}
        factUrl={this.props.planetFacts['Pluto'].factUrl}
        factSite={'Space.com'}
      /> } />
      </>
    )
  }
}

const mapStateToProps = state => ({
  planetFacts: state.planetFacts
});

export default connect(mapStateToProps)(App);
