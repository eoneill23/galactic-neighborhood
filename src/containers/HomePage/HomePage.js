import Asteroid from '../../components/Asteroid/Asteroid';
import SpaceStationInfo from '../../components/SpaceStationInfo/SpaceStationInfo';
import React, { Component } from 'react';
import { fetchAPOD, fetchISS, fetchAsteroids } from '../../util/apiCalls';
import { addAPOD, addAPODError, addISS, addISSError, addAsteroids, addAsteroidsError } from '../../actions/index';
import { connect } from 'react-redux';
import './HomePage.css';

class HomePage extends Component {
  constructor() {
    super();
    this.state = {
      error: ''
    }
  }

  async componentDidMount() {
    try {
      const APOD = await fetchAPOD()
      this.props.addAPOD(APOD)
    } catch (error) {
      this.props.addAPODError(error.message)
    }

    setInterval( async () => {
      try {
        const ISS = await fetchISS()
        this.props.addISS(ISS)
      } catch (error) {
        this.props.addISSError(error.message)
      }
    }, 2000)

    try {
      let formattedDate = new Date().toISOString().slice(0, 10);
      const asteroids = await fetchAsteroids(formattedDate)
      this.props.addAsteroids(asteroids.near_earth_objects[formattedDate])
    } catch (error) {
      console.log("IN THE ERROR", error)
      this.props.addAsteroidsError(error.message)
    }
  }

  render() {

    let asteroidsList;

    if (this.props.asteroids.length) {
      asteroidsList = this.props.asteroids.map(asteroid => {
        return <Asteroid
          key={asteroid.name}
          name={asteroid.name}
          minDiam={asteroid.estimated_diameter.miles.estimated_diameter_min.toFixed(3)}
          maxDiam={asteroid.estimated_diameter.miles.estimated_diameter_max.toFixed(3)}
          isHazardous={asteroid.is_potentially_hazardous_asteroid}
        />
      })
    } else {
      asteroidsList = <p>Getting asteroid info...</p>
    }

    return (
      <section className='home-container'>
        <div className='img-container'>
          <h2 className='img-header'>Today's astronomy picture of the day:</h2>
          {this.props.apodError && <p>{this.props.apodError}</p>}
          {this.props.APOD && <img src={this.props.APOD.url} className='apod' alt='The astronomy picture of the day'/>}
          {this.props.APOD && <p className='img-caption'>{this.props.APOD.title}.</p>}
          {this.props.APOD && this.props.APOD && <p className='img-copyright'>Â© {this.props.APOD.copyright} {this.props.APOD.date}</p>}
        </div>
        <section>
          <h2>Asteroids near Earth today:</h2>
          <ul className='asteroids-list'>
            {asteroidsList}
          </ul>
        </section>
        <div className='iss-container'>
          <h2>Where is the International Space Station?</h2>
          {this.props.issError && <p>{this.props.issError}</p>}
          {this.props.iss && 
            <SpaceStationInfo 
              lat={this.props.iss.latitude}
              long={this.props.iss.longitude}
              velo={this.props.iss.velocity.toFixed(2)}
              altitude={this.props.iss.altitude.toFixed(2)}
            />
          }
        </div>
      </section>
    )
  }
}

export const mapStateToProps = state => ({
  APOD: state.apod,
  apodError: state.apodError,
  iss: state.iss,
  issError: state.issError,
  asteroids: state.asteroids,
  asteroidsError: state.asteroidsError
});

export const mapDispatchToProps = dispatch => ({
  addAPOD: APOD => dispatch(addAPOD(APOD)),
  addAPODError: error => dispatch(addAPODError(error)),
  addISS: ISS => dispatch(addISS(ISS)),
  addISSError: error => dispatch(addISSError(error)),
  addAsteroids: asteroids => dispatch(addAsteroids(asteroids)),
  addAsteroidsError: error => dispatch(addAsteroidsError(error))
});

export default connect(mapStateToProps, mapDispatchToProps)(HomePage);