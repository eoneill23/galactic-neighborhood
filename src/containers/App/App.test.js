import React from 'react';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';
import { App, mapStateToProps } from './App';
import HomePage from '../HomePage/HomePage';

describe('App', () => {
  let wrapper, mockPlanetFacts;

  beforeEach(() => {
    mockPlanetFacts = {
      'Mercury': {
        fact: 'The closest planet to the Sun does indeed have ice on its surface. That sounds surprising at first glance, but the ice is found in permanently shadowed craters — those that never receive any sunlight. It is thought that perhaps comets delivered this ice to Mercury in the first place. In fact, NASA’s MESSENGER spacecraft not only found ice at the north pole, but it also found organics, which are the building blocks for life. Mercury is way too hot and airless for life as we know it, but it shows how these elements are distributed across the Solar System.',
        factUrl: 'https://www.universetoday.com/33415/interesting-facts-about-the-planets/',
        imgUrl: 'https://upload.wikimedia.org/wikipedia/commons/d/d9/Mercury_in_color_-_Prockter07-edit1.jpg'
      },
      'Venus': {
        fact: 'Both Mercury and Venus have no moons, which can be considered a surprise given there are dozens of other ones around the Solar System. Saturn has over 60, for example. And some moons are little more than captured asteroids, which may have been what happened with Mars’ two moons, for example. So what makes these planets different? No one is really sure why Venus doesn’t, but there is at least one stream of research that suggests it could have had one in the past.',
        factUrl: 'https://www.universetoday.com/33415/interesting-facts-about-the-planets/',
        imgUrl: 'https://upload.wikimedia.org/wikipedia/commons/thumb/8/85/Venus_globe.jpg/480px-Venus_globe.jpg'
      },
      'Earth': {
        fact: 'We have a magnetic field surrounding our planet that protects us from the blasts of radiation and particles the Sun sends our way. Good thing, too, because such flare-ups could prove deadly to unprotected people; that’s why NASA keeps an eye on solar activity for astronauts on the International Space Station, for example. At any rate, when you see auroras shining in the sky, that’s what happens when the particles from the Sun flow along the magnetic field lines and interact with Earth’s upper atmosphere.',
        factUrl: 'https://www.universetoday.com/33415/interesting-facts-about-the-planets/',
        imgUrl: 'https://upload.wikimedia.org/wikipedia/commons/9/97/The_Earth_seen_from_Apollo_17.jpg'
      }
    }

    wrapper = shallow(<App planetFacts={mockPlanetFacts} />)
  });

  it('should match the snapshot with all data passed in correctly', () => {

    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    const mockState = {
      apod: { url: 'https://apod.nasa.gov/apod/image/1909/HarvestmoonGraffand.jpg' },
      apodError: '',
      iss: { latitutde: 1, longitude: 2, velocity: 27000 },
      issError: '',
      asteroids: [
        {
          name: '(U2)',
          min_diameter: 2,
          max_diameter: 3
        },
        {
          name: '(K2)',
          min_diameter: 1,
          max_diameter: 4
        }
      ],
      asteroidsError: '',
      planetFacts: {
        'Mercury': {
          fact: 'The closest planet to the Sun does indeed have ice on its surface. That sounds surprising at first glance, but the ice is found in permanently shadowed craters — those that never receive any sunlight. It is thought that perhaps comets delivered this ice to Mercury in the first place. In fact, NASA’s MESSENGER spacecraft not only found ice at the north pole, but it also found organics, which are the building blocks for life. Mercury is way too hot and airless for life as we know it, but it shows how these elements are distributed across the Solar System.',
          factUrl: 'https://www.universetoday.com/33415/interesting-facts-about-the-planets/',
          imgUrl: 'https://upload.wikimedia.org/wikipedia/commons/d/d9/Mercury_in_color_-_Prockter07-edit1.jpg'
        }
      }
    }

    const expectedPlanetFacts = {
      'Mercury': {
        fact: 'The closest planet to the Sun does indeed have ice on its surface. That sounds surprising at first glance, but the ice is found in permanently shadowed craters — those that never receive any sunlight. It is thought that perhaps comets delivered this ice to Mercury in the first place. In fact, NASA’s MESSENGER spacecraft not only found ice at the north pole, but it also found organics, which are the building blocks for life. Mercury is way too hot and airless for life as we know it, but it shows how these elements are distributed across the Solar System.',
        factUrl: 'https://www.universetoday.com/33415/interesting-facts-about-the-planets/',
        imgUrl: 'https://upload.wikimedia.org/wikipedia/commons/d/d9/Mercury_in_color_-_Prockter07-edit1.jpg'
      }
    }

    expect(mapStateToProps(mockState).planetFacts).toEqual(expectedPlanetFacts);
  });

});

// describe('Route', () => {
//   it('should route to the home page by default', () => {

//     const wrapper = mount(
//       <MemoryRouter initialEntries={['/']}>
//         <App />
//       </MemoryRouter>
//     );

//     expect(wrapper.find(HomePage)).toHaveLength(1);
//   });
// });