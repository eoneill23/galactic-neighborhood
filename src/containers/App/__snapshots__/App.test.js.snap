// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<Fragment>
  <Header />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/mercury"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/venus"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/earth"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/mars"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/jupiter"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/saturn"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/uranus"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/neptune"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/pluto"
    render={[Function]}
  />
</Fragment>
`;
